File: ui.webapp/src/models/Company.js
import Entity from './Entity'

export default class Company extends Entity {
    constructor(id='',companyName='',fantasyName='',corporateNumber='') {
      super()
      this.id = id;
      this.companyName = companyName;
      this.fantasyName= fantasyName;
      this.corporateNumber=corporateNumber;
    }
  }

File: 0-Shared/Domain/Generals/Company.cs
using Framework.Core.Models;
using Newtonsoft.Json;

namespace Domain.Generals
{
    public class Company : Entity
    {
        [JsonProperty]
        public string CompanyName { get; protected set; }
        [JsonProperty]
        public string FantasyName { get; protected set; }
        [JsonProperty]
        public string CorporateNumber { get; protected set; }

        public bool Create(string companyName, string fantasyName, string corporateNumber,string by, string id = null)
        {
            CompanyName = companyName;
            FantasyName = fantasyName;
            CorporateNumber = corporateNumber;
            return base.Create(by, id);
        }

        public bool Update(string companyName, string fantasyName, string corporateNumber, string by)
        {
            CompanyName = companyName;
            FantasyName = fantasyName;
            CorporateNumber = corporateNumber;
            return base.Update(by);
        }
        
    }

}
